packages/hash/api/src/task-execution/index.ts

```diff
@@ -0,0 +1,36 @@
+import { DataSource } from "apollo-datasource";
+import fetch from "node-fetch";
+
+// TODO: When task scheduling is more mature and we move away from the temporary `hash-task-executor` we should have a single source of
+//  truth for available tasks, likely importable.
+export enum Tasks {
+  Demo = "python",
+}
+
+export type Config = {
+  host: string;
+  port: number;
+};
+
+export type TaskExecutor = ReturnType<typeof connectToTaskExecutor> &
+  DataSource;
+
+// A pretty dumb method at the moment, but provides a clean boundary where we can slot in logic for a more complex task-execution mechanism later on
+/**
+ *
+ * @param config The connection config for connecting to the Task-Executor
+ * @returns a connection object with a run_task method
+ */
+export const connectToTaskExecutor = (config: Config) => {
+  return {
+    run_task: async (route: Tasks) => {
+      try {
+        return await fetch(
+          `http://${config.host}:${config.port}/${route}`,
+        ).then((resp) => resp.json());
+      } catch (err: any) {
+        throw new Error(`Failed to execute task ${route}: ${err}`);
+      }
```

Comment:
Nit: could be simplified a bit by removing the `try {} catch {}` blocks
```suggestion
      return await fetch(`http://${config.host}:${config.port}/${route}`)
        .then((resp) => resp.json())
        .catch((error) => {
          throw new Error(`Failed to execute task ${route}: ${error}`);
        });
```